synapse/lib/node.py

```diff
@@ -529,7 +525,8 @@ async def _getTagDelEdits(self, tag, init=False):
             edits.append((s_layer.EDIT_TAG_DEL, (subtag, None), ()))
 
         edits.extend(self._getTagPropDel(name))
-        edits.append((s_layer.EDIT_TAG_DEL, (name, None), ()))
+        if self.tags.get(name, s_common.novalu) is not s_common.novalu:
```

Comment:
To prep for Syn 3.0 ( and make it easier for cisphyx to merge to the syn 3.0 branch ) please tweak to using `self.getTag(name, default=s_common.novalu)` here :)